sklearn/pipeline.py

```diff
@@ -257,6 +258,24 @@ def _log_message(self, step_idx):
                                                   len(self.steps),
                                                   name)
 
+    def get_props_request(self):
```

Comment:
There will be cases where the meta-estimator requests a prop, such as `sample_weight` regardless of whether its descendant estimators also request props.

The implementation here implies that the caller of `get_props_request` always wants the "deep" props of its descendants. Do we have use cases that would want us to be able to inspect only the locally consumed props?